stdlib/public/stubs/Availability.mm

```diff
@@ -22,72 +22,33 @@
 #import <Foundation/Foundation.h>
 #include <TargetConditionals.h>
 #include "../SwiftShims/FoundationShims.h"
-
-using namespace swift;
-
-/// Load the contents of the SystemVersion plist, looking in the appropriate
-/// place if running on the iOS simulator.
-static NSDictionary *systemVersionDictionaryFromPlist() {
-  NSString *plistPath = @"/System/Library/CoreServices/SystemVersion.plist";
-
-#if TARGET_OS_SIMULATOR
-  // When targeting the iOS simulator, look in a special location so we do
-  // not pick up the host OS version.
-  const char *simulatorRoot = getenv("IPHONE_SIMULATOR_ROOT");
-  if (!simulatorRoot) {
-    fatalError(/* flags = */ 0,
-               "Unable to check API availability: "
-               "IPHONE_SIMULATOR_ROOT not set when running under simulator");
-  }
-
-  plistPath = [NSString stringWithFormat:@"%s%@", simulatorRoot, plistPath];
-#endif
-
-  return [NSDictionary dictionaryWithContentsOfFile:plistPath];
+#import <objc/runtime.h>
+
+static NSOperatingSystemVersion getOSVersionFromFoundation() {
+  Class cls = objc_getClass("NSProcessInfo");
+  id pinfo = (id)[cls processInfo];
+  SEL sel = sel_registerName("operatingSystemVersion");
+  NSOperatingSystemVersion(*getvers)(id, SEL) =
+    (NSOperatingSystemVersion(*)(id, SEL))class_getMethodImplementation(cls, sel);
+  return getvers(pinfo, sel);
 }
 
-/// Load and parse the SystemVersion dictionary to determine the current
-/// operating system version.
-static NSOperatingSystemVersion operatingSystemVersionFromPlist() {
-  NSDictionary *plistDictionary = systemVersionDictionaryFromPlist();
-  if (!plistDictionary) {
-    fatalError(/* flags = */ 0,
-               "Unable to check API availability: "
-               "system version dictionary not found");
-  }
-
-  NSString *versionString = [plistDictionary objectForKey:@"ProductVersion"];
-  if (!versionString) {
-    fatalError(/* flags = */ 0,
-               "Unable to check API availability: "
-               "ProductVersion not present in system version dictionary");
+#if __has_include(<os/system_version.h>)
+#include <os/system_version.h>
+static NSOperatingSystemVersion getOSVersion() {
+  struct os_system_version_s vers;
+  if (os_system_version_get_current_version(&vers)) {
```

Comment:
I believe this should make availability checking significantly more efficient :)